idaes/core/control_volume1d.py

```diff
@@ -218,9 +225,22 @@ def add_geometry(self,
             self.area = Var(initialize=1.0,
                             doc='Cross-sectional area of Control Volume',
                             units=units("area"))
-        self.length = Var(initialize=1.0,
-                          doc='Length of Control Volume',
-                          units=units("length"))
+
+        if length_var is not None:
+            # Validate length_Var and add a reference
+            if not isinstance(length_var, (Var, Param, Expression)):
+                raise ConfigurationError(
+                    f"{self.name} length_var must be a Pyomo Var, Param or "
+                    "Expression.")
+            elif len(length_var) != 1:
+                raise ConfigurationError(
+                    f"{self.name} length_var must be a scalar (unidexed) "
+                    "component.")
+            add_object_reference(self, "length", length_var)
```

Comment:
Because `References` are always indexed, and we expect `length` to be uninxeded.

E.g.:

```
m.v = Var()
m.r = Reference(m.v)

m.c = Constraint(expr=42==m.r)  # this fails as m.r is indexed by [None]

```

Reply:
Yep, that's right. Slipped my mind.